code/game/objects/structures/crates_lockers/f13 closets.dm

```diff
@@ -16,34 +16,103 @@
 	new /obj/item/seeds/watermelon(src)
 	new /obj/item/seeds/poppy/broc(src)
 	new /obj/item/seeds/xander(src)
+	new /obj/item/seeds/wheat(src)
+	new /obj/item/seeds/cotton(src)
 	//new /obj/item/seeds/plantnamehrere(src)
 
-/obj/structure/closet/f13/leigon/scollar
+/obj/structure/closet/f13/leigon/slave
 	name = "closet"
 
-/obj/structure/closet/f13/leigon/scollar/PopulateContents()
+/obj/structure/closet/f13/leigon/slave/PopulateContents()
 	..()
-	new /obj/item/assembly/signaler/electropack/shockcollar(src)
-	new /obj/item/assembly/signaler/electropack/shockcollar(src)
-	new /obj/item/assembly/signaler/electropack/shockcollar(src)
-	new /obj/item/assembly/signaler/electropack/shockcollar(src)
+	for(var/i in 1 to 4)
+		new /obj/item/assembly/signaler/electropack/shockcollar(src)
+		new /obj/item/clothing/under/f13/caravan(src)
+		new /obj/item/clothing/shoes/sandal(src)
+	return
 
-/obj/structure/closet/f13/leigon/spants
-	name = "closet"
+/obj/structure/closet/cardboard/f13
+	name = "don't use this"
+
+/obj/structure/closet/cardboard/f13/legion
+	name = "still dont use this"
+
+/obj/structure/closet/cardboard/f13/legion/caravanshotgun
+	name = "large cardboard box"
 
-/obj/structure/closet/f13/leigon/spants/PopulateContents()
+/obj/structure/closet/cardboard/f13/legion/caravanshotgun/PopulateContents()
 	..()
-	new /obj/item/clothing/under/f13/caravan(src)
-	new /obj/item/clothing/under/f13/caravan(src)
-	new /obj/item/clothing/under/f13/caravan(src)
-	new /obj/item/clothing/under/f13/caravan(src)
+	for(var/i in 1 to 3)
+		new /obj/item/gun/ballistic/revolver/caravan_shotgun(src)
+	return
 
-/obj/structure/closet/f13/leigon/ssandals
-	name = "closet"
+/obj/structure/closet/cardboard/f13/legion/singleshotgun
+	name = "large cardboard box"
```

Comment:
You don't need to name = "large cardboard box" for each one, the name will auto-populate based on the parent due to being a var set higher up

Reply:
i'll like it to have it anyways for clearifiaction, like we don't need all the special same armour values for each copy when we can use subtypes. its not neccesary but its handy to do so.

Reply:
It leads to horribly unnecessary bloat, and leads to an issue where things are hard to find and sometimes you want all the children to change when you change the parent. Hard disagree on policy, long term convenience, and proper code management.

You can also write hierarchies as
/obj/item
                /parent
 parent procs defined here
                              /child
Etc, although I forget the exact details during my phone posting. I recommend getting away from the nondescriptive and nonuseful parents like f13 and making them all useful and applicable, so we don't have weird shenanigans where people spawn in weird things or map them incorrectly.

On a further note you may want to consider adding your own parents - maybe "loot" or "armory" or something and setting every faction and such to use the same derivatives. It would clean things up in the future and reduce the amount of work you have to put in too.